### ISSUE TITLE
[BUG] --tag not overiding package.json version

### ISSUE THREAD

**Issue by hardillb** [npm/cli#4685]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I have the following package.json

```json
{
  "name": "@test/test"
  "dependencies": {
    "@test/foo": "^0.3.3",
    "@test/bar": "^0.3.0"
  }
}
```
I have a private registry (verdaccio on npm.example.com) setup with the `@test` scope, in the registry there is following versions

 - @test/foo: 0.3.3 and 0.4.0-test.1
 - @test/bar: 0.3.0 and 0.4.0-test.1

The `0.3.x` releases are tagged with `latest` and the `0.4.0-test.1` releases are tagged as `beta`

Running `npm install --registry=https://npm.example.com --tag beta` installs the `0.3.x` not the `0.4.0-test.1`

From the npm install docs

> The --tag argument will apply to all of the specified install targets. If a tag with the given name exists, the tagged version is preferred over newer versions.

### Expected Behavior

With the `beta` tag added to the command line I expect `0.4.0-test.1` versions to be installed as they have the matching tag.

### Steps To Reproduce

1. In this environment...
2. With this config...
3. Run '...'
4. See error...

### Environment

- npm: 8.5.0
- Node.js: v16.14.1
- OS Name: Linux (Fedora 35)
- System Model Name: Think Pad X1 Carbon
- npm config:
```ini
; "global" config from /usr/etc/npmrc

prefix = "/usr/local" 
python = "/usr/bin/python3" 

; "user" config from /home/hardillb/.npmrc

//@test:registry = "https://npm.example.com/" 
//npm.example.com/:_authToken = (protected) 

; node bin location = /usr/bin/node
; cwd = /opt/share/projects/flowforge/test/ff
; HOME = /home/hardillb
; Run `npm config ls -l` to show all defaults.
```

**Comment by shadowspawn** [comment]
> The --tag argument will apply to all of the specified install targets.

It seems to work for targets listed explicitly on the command line. Try:

```
npm install --registry=https://npm.example.com --tag beta @test/foo @test/bar
```

**Comment by hardillb** [comment]
That might work, but it's not useful in this case.

I'm trying to pull in a bunch of dependencies at beta level in a container build, I can add the `--tag beta` into the command line in the container build, but I'm trying not to have a totally different `package.json` to pick up the normal release vs a beta release.

The best I've come up with that works is a `package.json` like this:

```json
{
  "name": "@test/test"
  "dependencies": {
    "@test/foo": "*",
    "@test/bar": "*"
  }
}
```

Which installs the `0.3.x` with  `npm install --registry=https://npm.example.com ` 

and  `0.4.0-test.1` with the `npm install --registry=https://npm.example.com  --tag beta`

**INSIGHT**
Just realised why this is happening. The `^` matches

> include everything that does not increment the first non-zero portion of semver
(https://semver.npmjs.com/)

So because the version starts with `0.` it's not matching because I am going from `0.3` to `0.4`. It would have worked if it had been `^1.3.x`

**Comment by lukekarrys** [comment]
@hardillb Does you latest update mean this is no longer an issue for you? I'm wondering if we could surface this better in the terminal logging. Something like `info No --tag=beta found for @test/bar@^0.3.0`

**Comment by lukekarrys** [comment]
Thanks @hardillb! I had that comment typed up yesterday and just sent it from a stale version of this page

**Comment by hardillb** [comment]
@lukekarrys I closed it because I now understood the current behaviour.

It doesn't appear possible to have multiple versions with the same tag e.g. both v0.3.0@beta and v0.4.0@beta? Which makes me think that a tag should override the version, but I can see changing the current behaviour is likely to be way more problematic.

Anyway... and yes some better logging would be nice (as always ;-))

**Comment by lukekarrys** [comment]
I agree on changing the default, but we at least want to make it less surprising when it doesn't work. I'm reopening to revisit the logging/documentation around this:

- [ ] `npm install --tag` should not silently pick a different tag
